### ISSUE TITLE
Visit-page-experience

### ISSUE THREAD

**Issue by kwk** [openshiftio/openshift.io#4030]
For almost a year I didn't report this issue but it is bugging me a lot everytime I go to the openshift start page.

When I go to https://openshift.io I get to see this page:

![bildschirmfoto von 2018-07-19 15-25-11](https://user-images.githubusercontent.com/193408/42945528-f00ab22e-8b68-11e8-93e5-b49fe5edd606.png)

Notice that you can see both, a Logout, a Login and a Signup button on this page. But the moment that page has fully rendered, I'm immediately redirected to the next page that looks completely different and only stays there for a few seconds:

![bildschirmfoto von 2018-07-19 15-25-12](https://user-images.githubusercontent.com/193408/42945529-f022a92e-8b68-11e8-8643-85a7d03b3a85.png)

This page looks completely unfinished and unpolished. But ups, I'm redirected again to this page:

![bildschirmfoto von 2018-07-19 15-25-17](https://user-images.githubusercontent.com/193408/42945530-f03d29ac-8b68-11e8-8a4f-3645c4881c74.png)

Here I always begin to think that I've finally reached my destination because my username is recognized and the logo appears, but wait this wasn't the final thing. Another redirection!

![bildschirmfoto von 2018-07-19 15-25-19](https://user-images.githubusercontent.com/193408/42945531-f058cb4e-8b68-11e8-9b5b-298b989af75d.png)

This is the final page that has a completely different logo than the one before.

Am I the only person who finds this annoying or is there an issue that already deals with this? Can't we do the redirection in the background somehow?

@joshuawilson @aslakknutsen @alexeykazakov @sbose78 does this have anything to do with how the auth works?

**Comment by aptmac** [comment]
I believe this is a duplicate of https://github.com/openshiftio/openshift.io/issues/2456 (open) and https://github.com/openshiftio/openshift.io/issues/2771 (closed as dupe of 2456)

**Comment by kwk** [comment]
@aptmac the issues share some details but mine is the only one that shows the whole 4 pages.

**Comment by sbose78** [comment]
Re: The multiple redirects after login

**Previously** we had a form in the `\_gettingstarted` page which checked if your Github & OpenShift Online account was connected. And if not: 

1. You had to *check*  "Connect my OpenShift Online" & "Connect my Github" accounts . 
2. Click "Connect".
3. get redirected to a page where you allowed ( tick marks in a checkbox ) OpenShift.io to manage your OpenShift Online account.
4. get redirected back to the `\_gettingstarted` page, and an ajax call would verify if OpenShift Online account & Github is connected.
5. Repeat steps above for connecting Github
6. get redirected back to the `\_gettingstarted` page, and an ajax call would verify if OpenShift Online account & Github is now connected. If yes, redirect to "_/home". 


We realized users needn't connect their Github account for using Planner, and also users found it confusing when being asked to explicitly connect their OpenShift Online account.

This is what happens **today**

1. We got rid of asking the user to connect Github in advance.
2. The UI automatically initiates the OAuth connection/process to connect OpenShift Online.
Since you are already logged in to RHD, you aren't asked to enter anything, and you return to `\_gettingstarted` Started page with your OpenShiftOnline account getting connected in the process, without asking the user to do anything.
3. Finding your OpenShift Online connected, you are sent to `/_home`

I acknowledge that multiple redirects can be confusing, but I wanted to explain why we have them here.

To solve this problem, we plan to 'authorize' the user to OpenShiftOnline entirely behind the scenes as a backend operation in future.

**Comment by kwk** [comment]
> To solve this problem, we plan to 'authorize' the user to OpenShiftOnline entirely behind the scenes as a backend operation in future.

Fair enough. Thanks for the detailed explanation @sbose78

**Comment by stevengutz** [comment]
Can we try to limit defect ownership to a single team at a time?  This one has three - which team should start looking at this?

**Comment by sbose78** [comment]
@stevengutz , keeping this assigned to `team/auth` and `team/ux` for now.

**Comment by kwk** [comment]
> Please find the UX recommendation:
> https://redhat.invisionapp.com/share/3AOH1QYH69G#/324483514_4030_Ux_Recomentation

@muruGanesan I've commented on the design. My main concern is that it is unclear to me how one gets from the "Start" to "Landing Page".

**Comment by muruGanesan** [comment]
@kwk, @sbose78, @aptmac, @serenamarie125,
Based on my discussion with Authentication team, I update the flow to address the issue.
NOTE: this is only a draft (wireframe) based on your review & feedback... I will update the screen with an appropriate visual design

https://redhat.invisionapp.com/share/3AOH1QYH69G#/324562585_4030_Login-2-Home

CC: @sunilmalagi, @AshishDurgude

**Comment by muruGanesan** [comment]
> > Based on my discussion with Authentication team, I update the flow to address the issue.
> 
> @muruGanesan will there be a redirect that is visible to the user, even for maybe a small amount of time? The visuals don't make sense to me.

@kwk, 
I agree with your points. 
Ideally, there should not be a delay in "login -to- home" flow. 
In other words, after login, users should be redirected  to "_home"  [without showing progress bar & 'gray-colour" screens]

As per my understanding and @sbose78 information(above), this may be addressed in upcoming trains...

**Comment by muruGanesan** [comment]
@kwk, @sbose78 
Please review the UX recommendation provide your inputs. If it satisfies the criteria, please close this issue.

**Comment by catrobson** [comment]
@muruGanesan @serenamarie125 I have left comments on the InVision.  

In most use cases, the user is already logged in, their log in has been cached.  I would expect two use cases here:

1. I am a logged out user of OpenShift.io (my log in has expired, or I manually logged out).  In this case, I would see the openshift.io marketing screen, and click log in.  This would follow the flow outlined in InVision so far.

2. I am a logged in user of OpenShift.io, returning to the OPenShift.io URL.  In this case, I would not expect to see the marketing page at all!  Think of any SaaS offering where you have a logged in view that is not the marketing view when you return to the URL.  InVision, Jira, Gmail, etc.  The expectation is that I am brought right into the product an my experience without seeing the marketing page at all.  I think the InVision file does not cover this use case, which is the most jarring use case outlined by this issue.

**Comment by muruGanesan** [comment]
@catrobson,  I got your points. Thanks!

I updated the flow for the scenario-2 (where users credentials are stored in the browser):
https://redhat.invisionapp.com/share/3AOH1QYH69G#/324562585_4030_Login-2-Home

**Comment by muruGanesan** [comment]
@kwk, 
Please review the UX recommendation provide your inputs (if any). 
https://redhat.invisionapp.com/share/3AOH1QYH69G#/324562585_4030_Login-2-Home

CC: @sbose78, @aptmac, @serenamarie125,

**Comment by sbose78** [comment]
@muruGanesan , Looks good.

However, we need to figure out how to handle the scenario where the user doesn't have openshift online connected. Currently, a couple of redirects which connects the user's openshift online to OSiO. Do we wanna keep it the same way? Generally, this needs to happen only once per user.

**Comment by muruGanesan** [comment]
@sbose78, 
IMO: This story is all about "user credentials are validated + stored in the Browser"
As per UX recommendation: we need to remove the progress bar, gray screens & reduce the waiting time.
Please refer: https://redhat.invisionapp.com/share/3AOH1QYH69G#/324562585_4030_Login-2-Home

Regarding 'First-time login' flow will remain unchanged.

**Comment by christianvogt** [comment]
@muruGanesan the invision links are not working for me.

**Comment by muruGanesan** [comment]
@christianvogt, 
Please try this link & let me know if you can access.
https://redhat.invisionapp.com/share/3AOH1QYH69G#/324562585_4030_Login-2-Home

**Comment by serenamarie125** [comment]
@christianvogt if you agree, can you add a link to the associated planner item, and we will close out.  Thanks!

**Comment by joshuawilson** [comment]
As long as this is a bug we will track it here. We just need to change the team. 

If we want to make it a Story then we will close this in favor of tracking in planner.

**Comment by christianvogt** [comment]
Links are working again. Must have been an issue with invision at the time as I kept getting errors.

**Comment by muruGanesan** [comment]
@christianvogt, thanks! 
@joshuawilson, if engineering team is going to handle this issue then I would like to remove UXD (i.e. my name) from the assignees' list.

**Comment by joshuawilson** [comment]
@muruGanesan that is the work flow that we follow. I had already changed the team from UX to UI. I just removed you as a user. You can do both of those when you are done with your work on the issue, that way someone else can pick it up.
